### PR TITLE
Support for KEY_catch_hook

### PR DESCRIPTION
**Pull request by eliangcs**
Allows to run hook trigger with `z.legacyScripting.run(bundle, 'trigger.hook', 'KEY')`.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -235,7 +236,23 @@ const legacyScriptingRunner = (Zap, zobj, app) => {
       } else {
         funcs.push(response => {
           response.throwForStatus();
-          return zobj.JSON.parse(response.content);
+          const data = zobj.JSON.parse(response.content);
+          if (!ensureArray) {
```
Comment (BrunoBernardino) [inline index.js]: I suppose this is what we'll use for the connection label call?

File: test/example-app/index.js [hunk 2]
```diff
@@ -82,6 +82,12 @@ const legacyScriptingSource = `
         contacts[0].name = 'Patched by KEY_pre_poll & KEY_post_poll!';
         return contacts;
       },
+
+      contact_hook_scripting_catch_hook: function(bundle) {
```
Comment (BrunoBernardino) [inline test/example-app/index.js]: We should probably add another test here for returning an `array`.

File: index.js (hunk 1 above)
Comment (eliangcs) [reply index.js]: Look for where we call `runEventCombo` with `ensureArray` being `false`. Currently, only `post_oauthv2_token` scripting method use this.

Comment (eliangcs) [reply index.js]: Sorry I was confused. Currently, this is used for OAuth2 to get and refresh access token.

File: test/example-app/index.js (hunk 2 above)
Comment (eliangcs) [reply test/example-app/index.js]: No problem! Added in eb0f462.

**Pull request merged**